cpp/oneapi/dal/algo/decision_forest/backend/cpu/train_kernel_cls.cpp

```diff
@@ -121,6 +121,35 @@ static result_t call_daal_kernel(const context_cpu& ctx,
             interop::convert_to_daal_homogen_table(arr_oob_per_obs_err, row_count, 1);
         daal_result.set(daal_df_cls_train::outOfBagErrorPerObservation, res_oob_per_obs_err);
     }
+
+    if (check_mask_flag(desc.get_error_metric_mode(),
+                        error_metric_mode::out_of_bag_error_accuracy)) {
+        auto arr_oob_accuracy_err = array<Float>::empty(1 * 1);
+        res.set_oob_err_accuracy(
+            dal::detail::homogen_table_builder{}.reset(arr_oob_accuracy_err, 1, 1).build());
+
+        const auto res_oob_accuracy_err =
+            interop::convert_to_daal_homogen_table(arr_oob_accuracy_err, 1, 1);
+        daal_result.set(daal_df_cls_train::outOfBagErrorAccuracy, res_oob_accuracy_err);
+    }
+
+    if (check_mask_flag(desc.get_error_metric_mode(),
+                        error_metric_mode::out_of_bag_error_decision_function)) {
+        auto arr_oob_decision_function_err =
```

Comment:
Please check for [the multiplication overflow](https://github.com/oneapi-src/oneDAL/blob/f8814b27954ab3ffa42a60de5e891016dcb2aee2/cpp/oneapi/dal/detail/common.hpp#L48).